[nrf fromlist] Add support for the nRF54H20 DK board

anangl opened the pull request:
Commits cherry picked from:
- https://github.com/zephyrproject-rtos/zephyr/pull/69747 (one commit brought as [nrf fromlist] because it has to be modified downstream due to hwmv2)
- https://github.com/zephyrproject-rtos/zephyr/pull/69924
- https://github.com/zephyrproject-rtos/zephyr/pull/69962
- https://github.com/zephyrproject-rtos/zephyr/pull/70062
- https://github.com/zephyrproject-rtos/zephyr/pull/70154
- https://github.com/zephyrproject-rtos/zephyr/pull/70172
- https://github.com/zephyrproject-rtos/zephyr/pull/70205
- https://github.com/zephyrproject-rtos/zephyr/pull/70411
- https://github.com/zephyrproject-rtos/zephyr/pull/70454
- https://github.com/zephyrproject-rtos/zephyr/pull/70491
- https://github.com/zephyrproject-rtos/zephyr/pull/70616

NordicBuilder commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_nordic | https://github.com/zephyrproject-rtos/hal_nordic/commit/5470822384781624efb2fda28cbc6a895a227677 | https://github.com/zephyrproject-rtos/hal_nordic/commit/13ac55b5b52c905642e9c54f069109d188aa5840 (master) | [zephyrproject-rtos/hal_nordic@54708223..13ac55b5](https://github.com/zephyrproject-rtos/hal_nordic/compare/5470822384781624efb2fda28cbc6a895a227677..13ac55b5b52c905642e9c54f069109d188aa5840) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

gmarull commented on boards/arm/nrf54h20dk_nrf54h20/board.cmake:
```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
+
+include(${ZEPHYR_BASE}/boards/common/nrfjprog.board.cmake)
```
pending removal

anangl replied:
removed

anangl commented:
Rebased.

anangl commented:
Added commits cherry-picked from https://github.com/zephyrproject-rtos/zephyr/pull/70411.
Replaced `[nrf fromlist]` commits with `[nrf fromtree]` where appropriate.

anangl commented:
The compliance checks fails in the same mysterious way as in https://github.com/zephyrproject-rtos/zephyr/pull/70411:
```
Error: See https://www.pylint.org/ for more details

 R0801:Similar lines in 2 files
File:scripts/west_commands/tests/test_nrf.py
Line:1
Column:0
 R0801:Similar lines in 2 files
File:scripts/west_commands/tests/test_nrf.py
Line:1
Column:0
```

gmarull commented:
> The compliance checks fails in the same mysterious way as in [zephyrproject-rtos/zephyr#70411](https://github.com/zephyrproject-rtos/zephyr/pull/70411):
> 
> ```
> Error: See https://www.pylint.org/ for more details
> 
>  R0801:Similar lines in 2 files
> File:scripts/west_commands/tests/test_nrf.py
> Line:1
> Column:0
>  R0801:Similar lines in 2 files
> File:scripts/west_commands/tests/test_nrf.py
> Line:1
> Column:0
> ```

please cherry-pick https://github.com/zephyrproject-rtos/zephyr/pull/70454 (and drop last 3 commits)

anangl commented:
> please cherry-pick [zephyrproject-rtos/zephyr#70454](https://github.com/zephyrproject-rtos/zephyr/pull/70454) (and drop last 3 commits)

done

anangl commented:
Rebased.

anangl commented:
Added commits from https://github.com/zephyrproject-rtos/zephyr/pull/70154.

anangl commented:
Rebased.

anangl commented:
Rebased.

anangl commented:
Rebased.
